lib/_http_server.js

```diff
@@ -362,10 +362,10 @@ function Server(options, requestListener) {
   if (typeof options === 'function') {
     requestListener = options;
     options = {};
-  } else if (options == null || typeof options === 'object') {
-    options = { ...options };
+  } else if (options == null) {
+    options = {};
```

Comment:
Are you sure this is a safe change? The previous code was creating a new object, which could not be mutated or referenced from user-land.

Reply:
Whoops, It looks like I need to create new object like this:

```js
else {
  validateObject(options, 'options');
  options = { ...options };
}
```